src/borg/testsuite/helpers.py

```diff
@@ -1207,3 +1210,25 @@ def test_passphrase_new_retries(self, monkeypatch):
 
     def test_passphrase_repr(self):
         assert "secret" not in repr(Passphrase("secret"))
+
+    def test_argon2(self, monkeypatch):
+        # Arrange
+        monkeypatch.setenv('BORG_PASSPHRASE', "hello, pass phrase")
+        p = Passphrase.new()
+
+        # Act
+        enc_key, mac_key = p.argon2(
+            salt=b'salt'*16,
+            time_cost=1,
+            memory_cost=4096,
+            parallelism=1,
+            type=argon2.low_level.Type.I
+        )
+
+        # Assert
+        assert len(enc_key)*8 == 256
+        assert len(mac_key)*8 == 256
+        # echo -n "hello, pass phrase" | argon2 saltsaltsaltsaltsaltsaltsaltsaltsaltsaltsaltsaltsaltsaltsaltsalt -i -t 1 -k 4096 -p 1 -l 64 -r
+        expected = 'b668a71dded9a9772686ab4c2da2a06c491087e9e27c14f9e798ad8b929506b43ef3878e4bb03055b6efbecddcdbfc58bfc1617a0e1790a46c1a9991c3269b34'
+        expected = b16decode(expected.upper())
+        enc_key + mac_key == expected
```

Comment:
besides the length checks, this is basically testing THEIR code.
can you check if they have own tests so we do not need to test it here?

Reply:
[It looks like they do have their own regression tests](https://github.com/hynek/argon2-cffi/blob/0c7b6ec66798b983a1966cbf8b802f8e9d68f1ad/tests/test_low_level.py#L130-L146). I'll remove this than.